login/service.go

```diff
@@ -152,6 +132,62 @@ func (gh *gitHubOAuth) Perform(ctx *app.AuthorizeLoginContext) error {
 	return ctx.TemporaryRedirect()
 }
 
+type identityError struct {
+	message interface{}
+	code    int
+}
+
+const (
+	noPrimaryEmail            int = 1
+	associatedUserNotFound    int = 2
+	unauthorizedGitHubRequest int = 3
+)
+
+func (gh *gitHubOAuth) getUserIdentity(ctx context.Context, ghtoken *oauth2.Token) (*account.Identity, *identityError) {
```

Comment:
Maybe time to rename this method, as oppose to what the name suggests, 'Get', It actually Does remote calls, creates, updates and gets Users and Identities.